fileseed.go

```diff
@@ -52,7 +57,27 @@ func (s *FileSeed) LongestMatchWith(chunks []IndexChunk) (int, SeedSegment) {
 			max = len(m)
 		}
 	}
-	return max, newFileSeedSegment(s.srcFile, match, s.canReflink, true)
+	if max > 0 {
+		src, err := os.Open(s.srcFile)
+		if err != nil {
+			return 0, nil, err
+		}
+		defer src.Close()
+		// We validate the segment here, there is no need to do it again when we write the data
+		segment = newFileSeedSegment(s.srcFile, match, s.canReflink, false)
+		if err = segment.validate(src); err != nil {
+			return 0, nil, err
+		}
```

Comment:
This will work, but I suspect it's going to have quite an impact on performance. That's because previously the validation (which requires open, read, hash, close) would only be done on the longest matches. With this change, it'll validate every possible seed even if it's not the longest and is never chosen to be used. It'll also do that many times over, even re-check chunks that were checked in previous iterations.

I believe, but haven't actually tried, that you can delay validation as before by making a change in seqencer.go instead. I'll leave a comment below.

Reply:
Right, I implemented your suggestion of moving the validation out in `sequencer.go` so that we just do the validation for the longest match.